### DISCUSSION TITLE
Job concurrency does not run latest HEAD

### DISCUSSION THREAD

**Discussion by stefanbuck** [community/community#26827] [Actions]
<p>Hi there,</p>
<p>I came across a weird behaviour (at least not as expected) when using the <code>concurrency</code> field. I’m using a fixed concurrency group like <code>concurrency: deploy-request</code>. The related workflow creates a new commit whenever the <code>repository_dispatch</code> event occurs.</p>
<p>If I now trigger two jobs in a row, the first job creates a new commit and the job passes. However, the latter seems to operate on the previous HEAD. The step <code>cat output.log</code> does not contain the value generated by the first job <a href="https://github.com/stefanbuck/workflow-concurrency-issue/runs/3651704899?check_suite_focus=true#step:3:4" rel="noopener nofollow ugc">see</a> and <a href="https://github.com/stefanbuck/workflow-concurrency-issue/runs/3651704899?check_suite_focus=true#step:5:12" rel="noopener nofollow ugc">git push</a> fails.</p>
<p>As a workaround I run <code>git pull origin main</code> right after the checkout</p><aside class="onebox githubblob" data-onebox-src="https://github.com/stefanbuck/workflow-concurrency-issue/blob/47bab81cd67f66ff84e18229f572e8d5768fea93/.github/workflows/deployment-request.yaml#L12">
  <header class="source">

      <a href="https://github.com/stefanbuck/workflow-concurrency-issue/blob/47bab81cd67f66ff84e18229f572e8d5768fea93/.github/workflows/deployment-request.yaml#L12" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/stefanbuck/workflow-concurrency-issue/blob/47bab81cd67f66ff84e18229f572e8d5768fea93/.github/workflows/deployment-request.yaml#L12" target="_blank" rel="noopener nofollow ugc">stefanbuck/workflow-concurrency-issue/blob/47bab81cd67f66ff84e18229f572e8d5768fea93/.github/workflows/deployment-request.yaml#L12</a></h4>

  <pre class="onebox">    <code class="lang-yaml">
      <ol class="start lines" start="2" style="counter-reset: li-counter 1 ;">
          <li>concurrency: deploy-request</li>
          <li>on:</li>
          <li>  repository_dispatch:</li>
          <li>    types: [deployment_request]</li>
          <li>jobs:</li>
          <li>  deploy:</li>
          <li>    name: Trigger deploy</li>
          <li>    runs-on: ubuntu-latest</li>
          <li>    steps:</li>
          <li>      - uses: actions/checkout@v2</li>
          <li class="selected">      - run: git pull origin main</li>
          <li>      - run: cat output.log</li>
          <li>      - run: cat $GITHUB_EVENT_PATH | jq --raw-output '.client_payload.message' &gt;&gt; output.log</li>
          <li>
          </li>
<li>      - name: Commit </li>
          <li>        shell: bash</li>
          <li>        run: |</li>
          <li>          git config --global user.email "${{ github.actor }}@users.noreply.github.com" &amp;&amp; \</li>
          <li>          git config --global user.name "${{ github.actor }}" &amp;&amp; \</li>
          <li>          git add ./output.log &amp;&amp; \</li>
          <li>          git commit -m 'Update file' &amp;&amp; \</li>
      </ol>
    </code>
  </pre>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Is this behaviour intentional / by design?</p>

**Comment by stefanbuck** [comment, marked as answer]
<p>I had another look and turns out that all is needed is to set the default branch as <code>ref</code> to make the checkout action pull the latest HEAD</p>
<pre><code class="lang-auto">- uses: actions/checkout@v2
  with: 
    ref: 'main'
</code></pre>
<p>From the <code>checkout</code> <a href="https://github.com/actions/checkout#usage" rel="noopener nofollow ugc">docs</a></p>
<blockquote>
<p>The branch, tag or SHA to checkout. When checking out the repository that triggered a workflow, this defaults to the reference or SHA for that event. Otherwise, uses the default branch.</p>
</blockquote>

**Comment by MarcelWilson** [comment]
<p>I am so glad you posted this.  I ran into the same issue thinking the default was to checkout HEAD already.</p>

**Comment by stefanbuck** [comment]
<p>Great to see that posting the solution was already helpful to you :grinning:</p>
